src/Infrastructure/Repository/TelemetryEventRepository.php

```diff
@@ -0,0 +1,193 @@
+     * @param bool $flush
+     *
+     * @return void
+     */
+    public function save(TelemetryEventInterface $telemetryEvent, bool $flush = true): void
+    {
+        $telemetryEvent = $this->telemetryEventMapper->mapToInfrastructureTelemetryEvent($telemetryEvent);
+
+        $this->getEntityManager()->persist($telemetryEvent);
+
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\Telemetry\TelemetryEventInterface $telemetryEvent
+     * @param bool $flush
+     *
+     * @throws \RuntimeException
+     *
+     * @return void
+     */
+    public function update(TelemetryEventInterface $telemetryEvent, bool $flush = true): void
+    {
+        $id = $telemetryEvent->getId();
+
+        if ($id === null) {
+            return;
+        }
+
+        /** @var \SprykerSdk\Sdk\Infrastructure\Entity\TelemetryEvent|null $infrastructureEntity */
+        $infrastructureEntity = $this->find($telemetryEvent->getId());
+
+        if ($infrastructureEntity === null) {
+            throw new RuntimeException('Unable to find telemetry event with id %s', $id);
+        }
+
+        $this->telemetryEventMapper->mapTelemetryEvents($telemetryEvent, $infrastructureEntity);
+
+        if ($flush) {
+            $this->getEntityManager()->flush();
+        }
+    }
+
+    /**
+     * @param int $maxAttemptsCount
+     * @param int $limit
+     * @param int $maxSyncTimestamp
+     *
+     * @return array<\SprykerSdk\SdkContracts\Entity\Telemetry\TelemetryEventInterface>
+     */
+    public function getTelemetryEvents(int $maxAttemptsCount, int $limit, int $maxSyncTimestamp): array
+    {
+        $expr = $this->getEntityManager()->getExpressionBuilder();
+
+        $telemetryEvents = $this->createQueryBuilder('te')
+            ->where($expr->lt('te.synchronizationAttemptsCount', ':maxAttemptsCount'))
+            ->andWhere(
+                $expr->orX(
+                    $expr->lt('te.lastSynchronisationTimestamp', ':maxSyncTimestamp'),
+                    $expr->isNull('te.lastSynchronisationTimestamp'),
+                ),
+            )
+            ->setMaxResults($limit)
+            ->setParameters([
+                'maxAttemptsCount' => $maxAttemptsCount,
+                'maxSyncTimestamp' => $maxSyncTimestamp,
+            ])
+            ->getQuery()
+            ->getResult();
+
+        return array_map([$this->telemetryEventMapper, 'mapToDomainTelemetryEvent'], $telemetryEvents);
+    }
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\Telemetry\TelemetryEventInterface $telemetryEvent
+     * @param bool $flush
+     *
+     * @throws \RuntimeException
+     *
+     * @return void
+     */
+    public function remove(TelemetryEventInterface $telemetryEvent, bool $flush = true): void
+    {
+        $telemetryEventId = $telemetryEvent->getId();
+
+        if ($telemetryEventId === null) {
```

Comment:
you don't need to call this method if telemetryEventId is null. and the same in others plass